flink-table/flink-sql-gateway-api/src/main/java/org/apache/flink/table/gateway/api/results/serde/LogicalTypeJsonSerializer.java

```diff
@@ -0,0 +1,210 @@
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.SerializerProvider;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+
+/**
+ * Json serializer for {@link LogicalType}.
+ *
+ * @see LogicalTypeJsonDeserializer for the reverse operation.
+ */
+@PublicEvolving
+public final class LogicalTypeJsonSerializer extends StdSerializer<LogicalType> {
+
+    private static final long serialVersionUID = 1L;
+
+    // --------------------------------------------------------------------------------------------
+    // Public string constants for serializer and deserializer
+    // --------------------------------------------------------------------------------------------
+
+    // Common fields
+    public static final String FIELD_NAME_TYPE_NAME = "type";
+    public static final String FIELD_NAME_NULLABLE = "nullable";
+
+    // CHAR, VARCHAR, BINARY, VARBINARY
+    public static final String FIELD_NAME_LENGTH = "length";
+
+    // DECIMAL, TIMESTAMP_WITHOUT_TIME_ZONE, TIMESTAMP_WITH_LOCAL_TIME_ZONE
+    public static final String FIELD_NAME_PRECISION = "precision";
+
+    // DECIMAL
+    public static final String FIELD_NAME_SCALE = "scale";
+
+    // MAP
+    public static final String FIELD_NAME_KEY_TYPE = "keyType";
+    public static final String FIELD_NAME_VALUE_TYPE = "valueType";
+
+    // ARRAY, MULTISET
+    public static final String FIELD_NAME_ELEMENT_TYPE = "elementType";
+
+    // ROW
+    public static final String FIELD_NAME_FIELDS = "fields";
+    public static final String FIELD_NAME_FIELD_NAME = "name";
+    public static final String FIELD_NAME_FIELD_TYPE = "fieldType";
+
+    // RAW
+    public static final String FIELD_NAME_CLASS = "class";
+    public static final String FIELD_NAME_SERIALIZER = "serializer";
+
+    public LogicalTypeJsonSerializer() {
+        super(LogicalType.class);
+    }
+
+    @Override
+    public void serialize(
+            LogicalType logicalType,
+            JsonGenerator jsonGenerator,
+            SerializerProvider serializerProvider)
+            throws IOException {
+        serializeInternal(logicalType, jsonGenerator);
+    }
+
+    private void serializeInternal(LogicalType logicalType, JsonGenerator jsonGenerator)
+            throws IOException {
+        jsonGenerator.writeStartObject();
+
+        // write common fields shared by all types
+        jsonGenerator.writeStringField(FIELD_NAME_TYPE_NAME, logicalType.getTypeRoot().name());
+        jsonGenerator.writeBooleanField(FIELD_NAME_NULLABLE, logicalType.isNullable());
+        // write special fields according to type root
+        switch (logicalType.getTypeRoot()) {
+            case BOOLEAN:
+            case TINYINT:
+            case SMALLINT:
+            case INTEGER:
+            case BIGINT:
+            case FLOAT:
+            case DOUBLE:
+            case DATE:
+            case NULL:
```

Comment:
NullType is alway nullable. I think we dont't need to add `"nullable": true` in the json.